its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java

```diff
@@ -996,22 +996,38 @@ void incrementalPrAnalysis_ProducesUnchangedFiles() throws IOException {
 
   @Test
   void checkMultiLanguageSupportWithSdkFormat() throws Exception {
-    if (!TestUtils.getMsBuildPath(ORCHESTRATOR).toString().contains("2019")) {
-      return; // This test is not supported on versions older than Visual Studio 2019 because the new SDK-style format was introduced with .NET Core.
-    }
-    var folderName = "MultiLanguageSupport";
-    BuildResult result = runBeginBuildAndEndForStandardProject(folderName, "");
+    assumeFalse(TestUtils.getMsBuildPath(ORCHESTRATOR).toString().contains("2017")); // new SDK-style format was introduced with .NET Core, we can't run .NET Core SDK under VS 2017 CI context
+    Path projectDir = TestUtils.projectDir(basePath, "MultiLanguageSupport");
+    String token = TestUtils.getNewToken(ORCHESTRATOR);
+    String folderName = projectDir.getFileName().toString();
+    // Begin step in MultiLanguageSupport folder
+    ScannerForMSBuild scanner = TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
+      .addArgument("begin")
+      .setProjectKey(folderName)
+      .setProjectName(folderName)
+      .setProjectVersion("1.0")
+      .setProperty("sonar.sourceEncoding", "UTF-8")
+      // Overriding environment variables to fallback to projectBaseDir detection
+      .setEnvironmentVariable("TF_BUILD_SOURCESDIRECTORY", "")
+      .setEnvironmentVariable("TF_BUILD_BUILDDIRECTORY", "")
+      .setEnvironmentVariable("BUILD_SOURCESDIRECTORY", "");
+    ORCHESTRATOR.executeBuild(scanner);
+    // Build solution inside MultiLanguageSupport/src folder
+    TestUtils.runMSBuild(ORCHESTRATOR, projectDir, "/t:Restore,Rebuild", "src/MultiLanguageSupport.sln");
+    // End step in MultiLanguageSupport folder
+    BuildResult result =  TestUtils.executeEndStepAndDumpResults(ORCHESTRATOR, projectDir, folderName, token);
     assertTrue(result.isSuccess());
 
-    // Outside.js is not detected: projectBaseDir is at .csproj level
-    // Excluded.js is excluded from the .csproj with the Remove attribute
+    // Files within the 'frontend' folder are not included in sonar-project.properties source/test
+    // Outside.js, Outside.sql are not detected: projectBaseDir is at .csproj level
+    // Excluded.js, Excluded.sql, Excluded.cs are excluded from the .csproj with the Remove attribute
     List<Issue> issues = TestUtils.allIssues(ORCHESTRATOR);
     assertThat(issues).hasSize(3)
       .extracting(Issue::getRule, Issue::getComponent)
       .containsExactlyInAnyOrder(
-        tuple("csharpsquid:S1134", folderName + ":" + folderName + "/Program.cs"),
-        tuple("javascript:S1529", folderName + ":" + folderName + "/JavaScript.js"),
-        tuple("plsql:S1134", folderName + ":" + folderName + "/plsql.sql"));
+        tuple("csharpsquid:S1134", folderName + ":Program.cs"),
```

Comment:
Aren't the folder names always the same after the change with env variables?

If the behavior and component names were hard coded, I think the code would be easier to understand.

Reply:
The folders are the same but the reporting is not:
**Before** (or with explicit projectBaseDir argument in the scanner)
`MultiLanguageSupport:MultiLanguageSupport/Program.cs
`**After** (with env. variables override)
`MultiLanguageSupport:Program.cs`

I'm not sure what happens tho if another `Program.cs` is found and analyzed in the `frontend` folder, for example 🤔 it's weird reporting indeed

Reply:
My understanding:
- this is a single project solution, and we do not explicitly set the `projectBaseDir` property. This means that the value of the property is automatically set to the folder that contains the csproj. Due to this, `Program.cs` is the identified path.
- if there will be another project, that will contain a Program.cs file, then the base directory will change to contain both projects, thus, the file paths will change too, to contain their parent path.

For now, I would hadcode the component names to "MultiLanguageSupport:", expecting to update the paths when the multi-language analysis feature is merged.

Reply:
Yes you are right, I don't know why but I had in mind the prototype scanner version with the projectBaseDir detected in the root folder where the scanner is called. Thanks for the clarification!